virttest/test_setup.py

```diff
@@ -317,6 +323,31 @@ def __init__(self, params):
 
         self.target_hugepages = target_hugepages
 
+    def check_hugepage_support(self):
+        """
+        Check whether the host support hugepage.
+        Need check the cpu flag and kernel CML.
+        """
+        error_context.context("Check whether the host support hugepage.")
+        utils_misc.check_cpu_flag(self.hugepage_cpu_flag)
+        self.check_hugepage_kernel_cmdline()
+
+    def check_hugepage_kernel_cmdline(self):
+        """
+        Check whether host kernel cmdline support hugepage support.
+        """
+        error_context.context("Check host kernel cmdline support hugepage")
+        utils_misc.check_kernel_cmdline(self.hugepage_match_str)
```

Comment:
return (self.hugepage_match_str in utils_misc.get_kernel_cmdline()
            and  self.hugepage_cpu_flag in utils_misc.get_cpu_flags() )